Using IoT Core as scalable ingest for hybrid projects

ptone opened the pull request.

dghale commented on tutorials/cloud-iot-hybrid/index.md:
```diff
@@ -0,0 +1,248 @@
+
+## Objectives
+
+- Configure IoT Core and Pubsub as a fully managed ingest system
```
%s/pubsub\/c/Cloud Pub\/Sub/

dghale commented on tutorials/cloud-iot-hybrid/index.md:
```diff
@@ -0,0 +1,248 @@
+
+Create a subscription to an MQTT topic of `sample`.
+
+Now you can use one of the [IoT Core quickstart samples](https://cloud.google.com/iot/docs/quickstart) to send test messages to the topic of 
```
...to the topic of:

dghale commented on tutorials/cloud-iot-hybrid/index.md:
```diff
@@ -0,0 +1,248 @@
+
+`/devices/${deviceId}/events/sample`
+
+You can also quickly emulate this by publishing directly to the pubsub topic directly with a `subFolder` attribute.
```
Cloud Pub/Sub

dghale commented on tutorials/cloud-iot-hybrid/index.md:
```diff
@@ -0,0 +1,248 @@
+This relay service could be adapted in a number of ways:
+
+- scaled out to consume more messages in parallel from pubsub - perhaps deployed to [Kubernetes Engine](https://cloud.google.com/kubernetes-engine/)
+- use rate limiting to protect the on-prem broker from overload, using the pubsub subscription as a "surge tank"
```
Capitalize list items and end each with a period.

dghale commented on tutorials/cloud-iot-hybrid/index.md:
```diff
@@ -0,0 +1,248 @@
+- scaled out to consume more messages in parallel from pubsub - perhaps deployed to [Kubernetes Engine](https://cloud.google.com/kubernetes-engine/)
+- use rate limiting to protect the on-prem broker from overload, using the pubsub subscription as a "surge tank"
+
+You might also consider having the relay pulling from Pubsub and writing more directly to alternate services on-prem, instead of through an on-prem MQTT broker. However by relaying MQTT - it lets the on-prem development proceed if MQTT was already built into the solution, or if a looser coupling is wanted.
```
Cloud Pub/Sub

replace dash with a comma: "...relaying MQTT, it lets the..."

dghale commented on tutorials/cloud-iot-hybrid/index.md:
````diff
@@ -0,0 +1,248 @@
+
+```sh
+gcloud beta compute instances create-with-container on-prem-rabbit \
+--zone=us-central1-c \
````
Could you use $CLOUD_ZONE here?

ptone commented:
Thanks @gguuss and @dghale  - I've incorporated your excellent feedback as well as some offline feedback from @chut

I ran through it end-end after resetting my cloud-shell to remove any lurking undocumented assumptions.

ptone commented:
@dghale I believe I addressed your comments, my understanding is to unblock the merge, I need an LGTM from you here on the PR - would love to get this merged next week

dghale commented:
LGTM

gguuss commented:
LGTM
